lib/src/analytics.dart

```diff
@@ -137,8 +194,64 @@ class AnalyticsImpl implements Analytics {
   @override
   bool get telemetryEnabled => _configHandler.telemetryEnabled;
 
+  @override
+  Map get userPropertyMap => userProperty.preparePayload();
+
+  @override
+  void close() => _gaClient.close();
+
+  @override
+  void sendEvent({
+    required DashEvents eventName,
+    required Map eventData,
+  }) {
+    if (!telemetryEnabled) return;
+
+    // Construct the body of the request
+    final Map body = {
+      'client_id': _clientId,
+      'events': [
+        {
+          'name': eventName,
+          'params': eventData,
+        }
+      ],
+      'user_properties': userProperty.preparePayload()
+    };
+
+    // Pass to the google analytics client to send
+    _gaClient.sendData(body);
+  }
+
   @override
   void setTelemetry(bool reportingBool) {
```

Comment:
To make this method async, do I only make the `setTelemetry` method async and leave the other methods within `ConfigHandler` as they are (ie. keep i/o related methods like `readAsStringSync()`, `lastModifiedSync()`, etc.) or do those need to also change to their async counterparts?

Is it enough to do this?
```dart

/// NEW UPDATE
Future<void> setTelemetry(bool reportingBool) async {
  _configHandler.setTelemetry(reportingBool);
}
```

```dart

/// PREVIOUS
void setTelemetry(bool reportingBool) {
  _configHandler.setTelemetry(reportingBool);
}
```

Reply:
you would need to make `_configHandler.setTelemetry` async, migrating readAsStringSync -> readAsString and writeAsStringSync -> writeAsString (I think lastModifiedSync is probably fast enough that it doesn't need to be async).

Also, did you mean to resolve this conversation?

Reply:
Ah, I marked it as resolved because I thought I figured it out by wrapping the config handler's method to set telemetry status in a future.

If I switch the read and write operations to async, wouldn't I have to `await` for them anyway since I need the contents of the file to toggle telemetry status?

Reply:
I have also refactored `setTelemetry` to the below code block
```dart
  @override
  Future<void> setTelemetry(bool reportingBool) {
    return Future<void>(() => _configHandler.setTelemetry(reportingBool));
  }
```

I tested it with the example file and it seems to be running asynchronously

Reply:
> I have also refactored `setTelemetry` to the below code block
> 
> ```dart
>   @override
>   Future<void> setTelemetry(bool reportingBool) {
>     return Future<void>(() => _configHandler.setTelemetry(reportingBool));
>   }
> ```
> 
> I tested it with the example file and it seems to be running asynchronously

This is technically asynchronous, however, the slow parts (reading and writing from/to disk) are still synchronous. If you don't want your PR to be blocked by this, you can assign a TODO to me and I will refactor in another PR.

Reply:
As discussed offline, the following issue https://github.com/eliasyishak/dash_analytics/issues/17 captures the work necessary for making this method async